incipyt/cli.py

```diff
@@ -0,0 +1,46 @@
+"""Command Line Interace functions."""
+
+import click
+import logging
+import pathlib
+
+from incipyt import actions, Environment, process_actions
+
+
+@click.command(help="incipyt is a command-line tool that bootstraps a Python project.")
+@click.argument(
+    "folder",
+    required=True,
+    default=pathlib.Path(),
+    type=click.Path(file_okay=False),
+    callback=lambda _ctx, _param, _path: pathlib.Path(_path),
+)
+@click.version_option()
+@click.option(
+    "--yes",
+    is_flag=True,
+    help="Do not ask confirmation for variables with a default value.",
+)
+@click.option(
+    "--check-build",
+    is_flag=True,
+    help="Build the package after initialization of all files and folders.",
+)
+def main(folder, yes, check_build):
+    logging.basicConfig(level="INFO")
+
+    env = Environment(auto_confirm=yes)
+    if folder == pathlib.Path():
+        if any(folder.resolve().iterdir()):
+            raise click.BadArgumentUsage(f"FOLDER {folder.resolve()} is not empty.")
```

Comment:
I would prompt for a confirmation instead, it is common to create a repository via github and initialize it with a LICENSE and a README. Same for the two `raise click.BadArgumentUsage("folder not empty")` below

Reply:
Right, but so we have to deal with the case of an already existing `.git` folder. Maybe for another PR and keep "empty folder requirement" meantime ?